concreteproperties/concrete_section.py

```diff
@@ -1258,6 +1293,7 @@ def micurve(progress=None):
                 # determine which index is the first to keep
                 if idx_to_keep == 0 and mi_res.n < max_comp:
                     idx_to_keep = idx
+                    break
 
             # create interpolation function and determine moment which corresponds to
```

Comment:
Rather than interpolating here, why not just calculate neutral axis depth `d_n` at `max_comp` and then work out actual moment capacity at this control point. 

I wonder if `max_comp` used, maybe easier to add another artificial control point at this axial load so you calculate the moment capacity value in above code iterations when going through all the points, then you're just cutting off the unused values above max_comp here?

This would be more accurate.

Reply:
This is a far more sensible approach, thanks. I was approaching this as a (bad) programmer not an engineer 🤣 

Before I implement this, just wanting to get clarification on something - should the first point in the diagram be at zero moment or have the same moment as the pretrimmed diagram, i.e. point 1 or 2 in the below? Current implementation is point 2.

![Screenshot 2022-11-07 130727](https://user-images.githubusercontent.com/18841910/200212123-40ee8886-eea0-4e10-845c-38244dabbf83.png)

Reply:
@robbievanleeuwen, I'd say zero moment (point 2), simply because if you plotted the negative side of the diagram it all meets up as a horizontal line anyway to complete the full interaction surface slice. So the actual value isn't that important as at that axial load you know you can have a moment between whatever the capacity is at `-M_max_comp` or `+M_max_comp`.

I'll have a look at what SPColumn does though to see if there is another precedent.

I was thinking that this additional point just gets appended to the evenly spaced points, rather than a control point so as not to mess up the labels?

I wonder if `max_comp` is specified if that should then substitute in for the upper limit so as not to waste computation time working out points above this axial load (don't recall maybe you already did this, been a week since I did the review!?). I'm thinking here for wall design in NZ at least, the max compression load is like `0.3f'cAg` which is usually way below the balance point, so there is no point working out data points above this only to then truncate the results and throw a large percentage of them away

Reply:
Ok thanks for that, some good points, will implement!

Reply:
Hi @robbievanleeuwen 

> I'll have a look at what SPColumn does though to see if there is another precedent.

I had a look at this but forgot to post back, what SPColumn does if you look at the positive or negative branches of the diagram is just cut it off at zero moment at the axes, but has the part of the negative curve generated at max compression or max tension, for example +ve and -ve views below:-
![image](https://user-images.githubusercontent.com/36808164/201494114-0770bc06-eaab-457f-a0f8-1969b7f9356c.png)
![image](https://user-images.githubusercontent.com/36808164/201494120-2b441958-fb98-421b-89f7-5a4f8d0a5d74.png)

So you get the full half of the diagram, not a skewed result for a non-symmetric section or reinforcement with different moment capacities top and bottom like concreteproperties returns. That could lead to not quite knowing for example if in your last diagram you're at an N*=-350kN and M*=25kNm that you're above or below the curve even though you're in the positive half of the interaction curve. 

As a thought I wonder how hard would it be to generate the remaining missing points back to zero moment in the tension region in your last example and to have the option of truncating the curve at zero moment to implement something similar to SPColumn for a consistent approach to a positive and negative portion of the curve. This would also make it easier to test if inside the curve or not if you added a closing line at zero moment and then used some shapely stuff to check if a point lies within a polygon or similar.

Reply:
Thanks for exploring this. I'm going to turn this into a new issue as it raises a few items and it would be nice to get this PR merged for now.